src/Chapter4.hs

```diff
@@ -629,7 +670,7 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM ma mb = ma >>= \a -> fmap (a &&) mb
```

Comment:
The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face:

However, your current solution will return `Nothing` on `andM (Just False) Nothing` instead of `Just False`.

Reply:
regarding `andM (Just False) Nothing`, I'd think this be similar to `>>=` where any Nothing in the chain would short the entire result

we also have `andM Nothing (Just False)` and don't think we can do anything to return `Just False` for this case?

Reply:
Well, existing `&&` is also lazy only on the right (second) argument. So the desired behaviour of `andM` can be consistent with `&&`:

```haskel
λ: False && undefined
False
λ: undefined && False
*** Exception: Prelude.undefined
CallStack (from HasCallStack):
  error, called at libraries/base/GHC/Err.hs:74:14 in base:GHC.Err
  undefined, called at <interactive>:6:1 in interactive:Ghci3
```

Reply:
thankyou that helps :)